smarts/core/agent_manager.py

```diff
@@ -112,7 +112,11 @@ def remove_pending_agent_ids(self, agent_ids):
         assert agent_ids.issubset(self.agent_ids)
         self._pending_agent_ids -= agent_ids
 
-    def observe_from(self, sim, vehicle_ids):
+    def observe_from(
+        self, sim, vehicle_ids: Set[str], done_this_step: Set[str] = set()
+    ) -> Tuple[
+        Dict[str, Observation], Dict[str, float], Dict[str, float], Dict[str, bool]
+    ]:
```

Comment:
from smarts.core.smarts import SMARTS
```suggestion
    def observe_from(
        self, sim: SMARTS, vehicle_ids: Set[str], done_this_step: Set[str] = set()
    ) -> Tuple[
        Dict[str, Observation], Dict[str, float], Dict[str, float], Dict[str, bool]
    ]:
```

Reply:
fwiw, I sort of disagree with this one for complicated reasons, although I'm willing to do it.

Reply:
So I tried that (with the import as you specified it) but got a weird error on importing due to circular references:
```
Traceback (most recent call last):
  File "/home/steve/Repos/Huawei/SMARTS/examples/observation_collection_for_imitation_learning.py", line 9, in <module>
    from smarts.core.smarts import SMARTS
  File "/home/steve/Repos/Huawei/SMARTS/smarts/core/smarts.py", line 42, in <module>
    from .agent_manager import AgentManager
  File "/home/steve/Repos/Huawei/SMARTS/smarts/core/agent_manager.py", line 31, in <module>
    from smarts.core.smarts import SMARTS
ImportError: cannot import name 'SMARTS' from 'smarts.core.smarts' (/home/steve/Repos/Huawei/SMARTS/smarts/core/smarts.py)
```

It's probably possible to fix this, but given that I'm already a bit predisposed against specifying `SMARTS` here as the type, I'm going to go ahead and pass on this one.